src/common/lib/config_storage.c

```
@@ -75,16 +109,18 @@
+    .version        = CONFIG_VERSION_CURRENT,
+    .crc16          = 0,  // Calculated at save time
+    .sequence       = 0,
+    .log_level      = LOG_LEVEL_DEFAULT,       // From config.h
     .led_brightness = LED_BRIGHTNESS_DEFAULT,  // From config.h or default
-    .flags = { .dirty = 0, .reserved = 0 },
-    .reserved = {0},
-    .storage = {0}
-};
+    .keyboard_id    = 0,                       // Set dynamically at init (from get_keyboard_id())
+    .flags          = {.dirty = 0, .shift_override_enabled = 0, .reserved = 0},
+    .layer_state    = 0x01,        // Layer 0 always active (bit 0 set)
```

Cppcheck (2.19.0)

[information] Include file

(missingIncludeSystem)